pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -264,6 +264,10 @@ func (h *AppHandler) collectHealthStatus(ctx context.Context, wl *appfile.Worklo
 
 	var traitStatusList []common.ApplicationTraitStatus
 	for _, tr := range wl.Traits {
+		if tr.FullTemplate.TraitDefinition.Spec.ControlPlaneOnly {
+			namespace = appRev.GetNamespace()
```

Comment:
where is the namespace used?

Reply:
> where is the namespace used?

Whe `ControlPlaneOnly=true`, the trait CR will be dispatched to the `local` cluster and the namespace that is same as application

Reply:
got